From 4eb3ab95c75eb4eb33c01d22affdfd303590d87a Mon Sep 17 00:00:00 2001
From: Haidong Yu <yuhaidong@uniontech.com>
Date: Mon, 21 Sep 2020 17:55:35 +0800
Subject: [PATCH] conhost: Add missing LeaveCriticalSection.

Signed-off-by: Haidong Yu <yuhaidong@uniontech.com>
Signed-off-by: Alexandre Julliard <julliard@winehq.org>
Signed-off-by: Wei Xie <xiewei@deepin.com>
---
 programs/conhost/conhost.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/programs/conhost/conhost.c b/programs/conhost/conhost.c
index c637b163795..c0a0e08e6d9 100644
--- a/programs/conhost/conhost.c
+++ b/programs/conhost/conhost.c
@@ -1659,6 +1659,7 @@ static DWORD WINAPI tty_input( void *param )
             switch (ch)
             {
             case 3: /* end of text */
+                LeaveCriticalSection( &console_section );
                 return 0;
             case '\n':
                 key_press( console, '\n', VK_RETURN, LEFT_CTRL_PRESSED );
-- 
2.20.1

