From 2697b74d6753a4bf528659047c2ce971f39fb6e5 Mon Sep 17 00:00:00 2001
From: Haoyang Chen <chenhaoyang@uniontech.com>
Date: Fri, 9 Oct 2020 10:24:54 +0800
Subject: [PATCH] gdiplus: Fix texture brush rotation.

Signed-off-by: Haoyang Chen <chenhaoyang@uniontech.com>
Signed-off-by: Esme Povirk <esme@codeweavers.com>
Signed-off-by: Alexandre Julliard <julliard@winehq.org>
Signed-off-by: Wei Xie <xiewei@deepin.com>
---
 dlls/gdiplus/graphics.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/gdiplus/graphics.c b/dlls/gdiplus/graphics.c
index 1615f592d51..190f45d64a2 100644
--- a/dlls/gdiplus/graphics.c
+++ b/dlls/gdiplus/graphics.c
@@ -1374,7 +1374,7 @@ static GpStatus brush_fill_pixels(GpGraphics *graphics, GpBrush *brush,
                 {
                     GpPointF point;
                     point.X = draw_points[0].X + x * x_dx + y * y_dx;
-                    point.Y = draw_points[0].Y + y * x_dy + y * y_dy;
+                    point.Y = draw_points[0].Y + x * x_dy + y * y_dy;
 
                     argb_pixels[x + y*cdwStride] = resample_bitmap_pixel(
                         &src_area, fill->bitmap_bits, bitmap->width, bitmap->height,
-- 
2.20.1

